src/platform/EFR32/DiagnosticDataProviderImpl.cpp

```diff
@@ -274,5 +341,143 @@ void DiagnosticDataProviderImpl::ReleaseNetworkInterfaces(NetworkInterface * net
     }
 }
 
+#if SL_WIFI
+CHIP_ERROR DiagnosticDataProviderImpl::GetWiFiBssId(ByteSpan & BssId)
+{
+    wfx_wifi_scan_result_t ap;
+    int32_t err = wfx_get_ap_info(&ap);
+    if (err == 0)
+    {
+        BssId = ByteSpan(ap.bssid, 6);
```

Comment:
This is handing out a pointer to an on-stack data structure, no?  How is that supposed to work?

Reply:
Sorry, didn't understand what you meant. It gets the details via the platform call and update the value ap and returns the required value accordingly

Reply:
It's handing the caller the `ap.bssid` value.  That's a pointer, right?  What is that pointer pointing to?

Reply:
Got it, resolved this in the latest commit